web/html/src/components/formulas/FormulaComponentGenerator.tsx

```diff
@@ -331,6 +350,29 @@ function checkVisibilityCondition(id, condition, formulaForm) {
   return false;
 }
 
+export function isFiltered(criteria) {
+  return criteria && criteria.length > 0;
+}
+
+// return the element content visibility conditionally based on the element name by the criteria
+function isVisibleByCriteria(element: any, criteria: string) {
+  let visibilityForcedByChildren = false;
+  // check if all children are not visible by criteria so we can hide the parent (this element) as well
+  for (var child_name in element) {
+    if (child_name.startsWith("$")) continue;
```

Comment:
This could use a comment please since it isn't clear what this special case is for.

Reply:
IIRC this is the condition [already in use](https://github.com/uyuni-project/uyuni/blob/master/web/html/src/components/formulas/FormulaComponentGenerator.tsx#L337) around in the codebase to check whether child_elements are meant to be `nested components` or simple items or something else, and we want to apply the _search and filter_ functionality only on top of `nested components`. The ones who starts with `$` are normal labels or other values, so we want to skip them.

Here it is an example of how a JSON formula content is served to the page:
```
{
  "formula_list": [
    "bind"
  ],
  "layout": {
    "bind": {
      "$type": "hidden-group",
      "config": {
        "$type": "group",
        "options": {
          "$type": "edit-group",
          "$optional": true,
          "$prototype": {
            "$type": "text",
            "$key": {
              "$type": "text",
              "$name": "Option"
            }
          }
        },
        "include_forwarders": {
          "$type": "boolean",
          "$default": false
        }
      },
      "configured_zones": {
        "$type": "edit-group",
        "$minItems": 1,
        "$itemName": "Zone ${i}",
        "$prototype": {
          "$type": "group",
          "$key": {
            "$type": "text",
            "$name": "Name"
          },
          "type": {
            "$type": "select",
            "$values": [
              "master",
              "slave"
            ],
            "$default": "master"
          },
          "notify": {
            "$type": "boolean",
            "$default": false
          }
        }
      },
      "available_zones": {
        "$type": "edit-group",
        "$minItems": 1,
        "$itemName": "Zone ${i}",
        "$prototype": {
          "$type": "group",
          "$key": {
            "$type": "text",
            "$name": "Name"
          },
          "file": {
            "$type": "text"
          },
          "soa": {
            "$name": "SOA",
            "$type": "group",
            "ns": {
              "$name": "NS",
              "$type": "text",
              "$placeholder": "ns@zone",
              "$ifEmpty": "ns"
            },
            "contact": {
              "$type": "text",
              "$placeholder": "admin@domain",
              "$ifEmpty": "root@localhost"
            },
            "serial": {
              "$default": "auto",
              "$ifEmpty": "auto"
            },
            "class": {
              "$default": "IN"
            },
            "refresh": {
              "$default": 8600,
              "$type": "number"
            },
            "retry": {
              "$default": 900,
              "$type": "number"
            },
            "expiry": {
              "$default": 86000,
              "$type": "number"
            },
            "nxdomain": {
              "$name": "NXDOMAIN",
              "$default": 500,
              "$type": "number"
            },
            "ttl": {
              "$name": "TTL",
              "$default": 8600,
              "$type": "number"
            }
          },
          "records": {
            "$type": "group",
            "A": {
              "$type": "edit-group",
              "$optional": true,
              "$minItems": 0,
              "$prototype": {
                "$key": {
                  "$type": "text",
                  "$name": "Hostname"
                },
                "$type": "text",
                "$name": "IP address"
              }
            },
            "NS": {
              "$name": "NS",
              "$type": "group",
              "$optional": true,
              "@": {
                "$type": "edit-group",
                "$minItems": 0,
                "$prototype": {
                  "$type": "text"
                }
              }
            },
            "CNAME": {
              "$name": "CNAME",
              "$type": "edit-group",
              "$optional": true,
              "$minItems": 0,
              "$prototype": {
                "$key": {
                  "$type": "text",
                  "$name": "Alias"
                },
                "$type": "text",
                "$name": "Hostname"
              }
            }
          },
          "generate_reverse": {
            "$type": "group",
            "$optional": true,
            "net": {
              "$name": "Network",
              "$optional": true
            },
            "for_zones": {
              "$type": "edit-group",
              "$optional": true,
              "$minItems": 0,
              "$prototype": {
                "$type": "text"
              }
            }
          }
        }
      }
    }
  },
  "system_data": {},
  "metadata": {
    "description": "Settings for bind nameserver",
    "group": "general_system_configuration",
    "after": [
      "branch-network"
    ]
  },
  "group_data": {},
  "formula_name": "bind"
}
```

@parlt91 feel free to pick any of the sentence from this comment and adapt it to add a comment, I think @Etheryte 's point is pretty valid, indeed I had to lookup again what and why I wrote it like that. Sorry for missing to write a clear comment since the beginning.